From ed3f92c12d9187e999f73991e9e5a77c104be26d Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Wed, 30 Sep 2015 16:15:00 +0800
Subject: [PATCH 0525/5242] MLK-11657: IPU: check ldb clock with clk name

commit  c4e3bdc0b3d94f1bb3a16bbac2e7942320ab1e9e from
https://source.codeaurora.org/external/imx/linux-imx.git

4.1 kernel apply the followed patch:
commit 73e0e496afdac9a5190eb3b9c51fdfebcc14ebd4
clkdev: Always allocate a struct clk and call __clk_get() w/ CCF

clock_get will return a new struct clk, so we can't use the pointer
of clk struct to compare clk whether is equal, replace with clk name.

Signed-off-by: Sandor Yu <R01008@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/ipu3/ipu_disp.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/ipu3/ipu_disp.c b/drivers/mxc/ipu3/ipu_disp.c
index f7044c2..633fc56 100644
--- a/drivers/mxc/ipu3/ipu_disp.c
+++ b/drivers/mxc/ipu3/ipu_disp.c
@@ -1086,8 +1086,8 @@ int32_t ipu_init_sync_panel(struct ipu_soc *ipu, int disp, uint32_t pixel_clk,
 		dev_err(ipu->dev, "clk_get di1 failed");
 		return PTR_ERR(ldb_di1_clk);
 	}
-
-	if (ldb_di0_clk == di_parent || ldb_di1_clk == di_parent) {
+	if (!strcmp(__clk_get_name(di_parent), __clk_get_name(ldb_di1_clk)) ||
+		!strcmp(__clk_get_name(di_parent), __clk_get_name(ldb_di1_clk))) {
 		/* if di clk parent is tve/ldb, then keep it;*/
 		dev_dbg(ipu->dev, "use special clk parent\n");
 		ret = clk_set_parent(ipu->pixel_clk_sel[disp], ipu->di_clk[disp]);
-- 
1.7.9.5

